x/stake/keeper/delegation.go

```diff
@@ -28,35 +28,108 @@ func (k Keeper) GetAllDelegations(ctx sdk.Context) (delegations []types.Delegati
 	iterator := sdk.KVStorePrefixIterator(store, DelegationKey)
 
 	i := 0
-	for ; ; i++ {
-		if !iterator.Valid() {
-			break
-		}
+	for ; iterator.Valid(); iterator.Next() {
```

Comment:
`for i := 0; iterator.Valid() && i < int(maxRetrieve[0]); iterator.Next() {`

Reply:
I took it off since it's not used in the function